### PR TITLE
Add Infra for porting lib only packages to use dotnet pack instead of pkgprojs

### PR DESCRIPTION
**Pull request by Anipik**
Related PR https://github.com/dotnet/runtime/pull/48385

Design Changes
- no more pkgproj
- separate symbol nuspec files
- minor change in metadata like no owners (Metadata not supported by the nuget pack)
- nuspec name contains package version
- no more pkgpath files (not sure why we were producing it in the first place)
- no more package reports
- removed implicit framework assembly references from the package like mscorlib.dll for net472
- exclude="Build,Analyzers"  for all dependencies

Validation changes
- Closure verification in the all config
- Package validation during build packages is no longer done as it was not doing any substantial work for extensions packages.

Test
- verified the nuspec files to check
- did a beynd comparer for the overall artifacts, changes were expected

- moving just one proj to unblock this effort and to add infra, (will start moving similar projs after this one is merged) 

Added few comments in response to the comments to the previous linked pr

### CODE REVIEW

File: src/libraries/libraries-packages.proj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.Build.Traversal" DefaultTargets="Pack">
 
   <PropertyGroup>
-    <TraversalGlobalProperties>BuildAllProjects=true</TraversalGlobalProperties>
+    <TraversalGlobalProperties>BuildAllProjects=true;BuildTargetFramework=</TraversalGlobalProperties>
```
Comment (Anipik) [inline src/libraries/libraries-packages.proj:4]: This one is required to enable building of packages after just a vertical build. i.e.

build -s clr.corelib+libs
dotnet build libraries-packages.proj /p:BuildAllConfigurations=true

becuase if buildtargetFramework is set then we wont build the all the configurations of the package.

**Comment by Anipik** [comment]
> Curious, why is <Serviceable>true</Serviceable> removed in the nuspec? Arcade sets it to true: https://github.com/dotnet/arcade/blob/99b9f805da070d301d76eac578a46dd75d9e5f7c/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectDefaults.props#L14. Is it omitted in case it's true?

not sure but its being passed as true to the pack task.

**Comment by Anipik** [comment]
```diff
@@ -1,15 +1,12 @@
 <EF><BB><BF><?xml version="1.0" encoding="utf-8"?>
-<package xmlns="http://schemas.microsoft.com/packaging/2013/01/nuspec.xsd">
-  <metadata minClientVersion="2.12">
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
     <id>Microsoft.Extensions.Configuration.EnvironmentVariables</id>
     <version>6.0.0-dev</version>
-    <title>Microsoft.Extensions.Configuration.EnvironmentVariables</title>
     <authors>Microsoft</authors>
-    <owners>microsoft,dotnetframework</owners>
-    <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <license type="expression">MIT</license>
+    <licenseUrl>https://licenses.nuget.org/MIT</licenseUrl>
     <icon>Icon.png</icon>
-    <projectUrl>https://dot.net/</projectUrl>
     <iconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</iconUrl>
     <description>Environment variables configuration provider implementation for Microsoft.Extensions.Configuration.</description>
     <releaseNotes>https://go.microsoft.com/fwlink/?LinkID=799421</releaseNotes>
@@ -18,27 +15,24 @@
     <repository type="git" url="https://github.com/dotnet/runtime" commit="0000000000000000000000000000000000000000" />
     <dependencies>
       <group targetFramework=".NETFramework4.6.1">
-        <dependency id="Microsoft.Extensions.Configuration" version="6.0.0-dev" />
-        <dependency id="Microsoft.Extensions.Configuration.Abstractions" version="6.0.0-dev" />
+        <dependency id="Microsoft.Extensions.Configuration.Abstractions" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="Microsoft.Extensions.Configuration" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
       <group targetFramework=".NETStandard2.0">
-        <dependency id="Microsoft.Extensions.Configuration" version="6.0.0-dev" />
-        <dependency id="Microsoft.Extensions.Configuration.Abstractions" version="6.0.0-dev" />
+        <dependency id="Microsoft.Extensions.Configuration.Abstractions" version="6.0.0-dev" exclude="Build,Analyzers" />
+        <dependency id="Microsoft.Extensions.Configuration" version="6.0.0-dev" exclude="Build,Analyzers" />
       </group>
     </dependencies>
-    <frameworkAssemblies>
-      <frameworkAssembly assemblyName="mscorlib" targetFramework=".NETFramework4.6.1" />
-    </frameworkAssemblies>
   </metadata>
   <files>
-    <file src="C:\Users\anagniho\.nuget\packages\microsoft.dotnet.arcade.sdk\6.0.0-beta.21219.2\tools\Assets\DotNetPackageIcon.png" target="Icon.png" exclude="" />
-    <file src="C:\git\runtime3\artifacts\bin\Microsoft.Extensions.Configuration.EnvironmentVariables\net461-Debug\Microsoft.Extensions.Configuration.EnvironmentVariables.dll" target="lib\net461" exclude="" />
-    <file src="C:\git\runtime3\artifacts\bin\Microsoft.Extensions.Configuration.EnvironmentVariables\net461-Debug\Microsoft.Extensions.Configuration.EnvironmentVariables.pdb" target="lib\net461" exclude="" />
-    <file src="C:\Users\anagniho\.nuget\packages\microsoft.private.intellisense\5.0.0-preview-20201009.2\IntellisenseFiles\net\1033\Microsoft.Extensions.Configuration.EnvironmentVariables.xml" target="lib\net461" exclude="" />
-    <file src="C:\git\runtime3\artifacts\bin\Microsoft.Extensions.Configuration.EnvironmentVariables\netstandard2.0-Debug\Microsoft.Extensions.Configuration.EnvironmentVariables.dll" target="lib\netstandard2.0" exclude="" />
-    <file src="C:\git\runtime3\artifacts\bin\Microsoft.Extensions.Configuration.EnvironmentVariables\netstandard2.0-Debug\Microsoft.Extensions.Configuration.EnvironmentVariables.pdb" target="lib\netstandard2.0" exclude="" />
-    <file src="C:\Users\anagniho\.nuget\packages\microsoft.private.intellisense\5.0.0-preview-20201009.2\IntellisenseFiles\net\1033\Microsoft.Extensions.Configuration.EnvironmentVariables.xml" target="lib\netstandard2.0" exclude="" />
-    <file src="C:\git\runtime3\LICENSE.TXT" target="LICENSE.TXT" exclude="" />
-    <file src="C:\git\runtime3\THIRD-PARTY-NOTICES.TXT" target="THIRD-PARTY-NOTICES.TXT" exclude="" />
+    <file src="C:\git\runtime4\artifacts\bin\Microsoft.Extensions.Configuration.EnvironmentVariables\net461-Debug\Microsoft.Extensions.Configuration.EnvironmentVariables.pdb" target="lib\net461\Microsoft.Extensions.Configuration.EnvironmentVariables.pdb" />
+    <file src="C:\git\runtime4\artifacts\bin\Microsoft.Extensions.Configuration.EnvironmentVariables\netstandard2.0-Debug\Microsoft.Extensions.Configuration.EnvironmentVariables.pdb" target="lib\netstandard2.0\Microsoft.Extensions.Configuration.EnvironmentVariables.pdb" />
+    <file src="C:\git\runtime4\artifacts\bin\Microsoft.Extensions.Configuration.EnvironmentVariables\net461-Debug\Microsoft.Extensions.Configuration.EnvironmentVariables.dll" target="lib\net461\Microsoft.Extensions.Configuration.EnvironmentVariables.dll" />
+    <file src="C:\git\runtime4\artifacts\bin\Microsoft.Extensions.Configuration.EnvironmentVariables\netstandard2.0-Debug\Microsoft.Extensions.Configuration.EnvironmentVariables.dll" target="lib\netstandard2.0\Microsoft.Extensions.Configuration.EnvironmentVariables.dll" />
+    <file src="C:\Users\anagniho\.nuget\packages\microsoft.private.intellisense\5.0.0-preview-20201009.2\IntellisenseFiles\net\1033\Microsoft.Extensions.Configuration.EnvironmentVariables.xml" target="lib\net461\Microsoft.Extensions.Configuration.EnvironmentVariables.xml" />
+    <file src="C:\Users\anagniho\.nuget\packages\microsoft.private.intellisense\5.0.0-preview-20201009.2\IntellisenseFiles\net\1033\Microsoft.Extensions.Configuration.EnvironmentVariables.xml" target="lib\netstandard2.0\Microsoft.Extensions.Configuration.EnvironmentVariables.xml" />
+    <file src="C:\Users\anagniho\.nuget\packages\microsoft.dotnet.arcade.sdk\6.0.0-beta.21219.2\tools\Assets\DotNetPackageIcon.png" target="Icon.png" />
+    <file src="C:\git\runtime4\LICENSE.TXT" target="LICENSE.TXT" />
+    <file src="C:\git\runtime4\THIRD-PARTY-NOTICES.TXT" target="THIRD-PARTY-NOTICES.TXT" />
   </files>
 </package>
\ No newline at end of file
```

**Comment by safern** [comment]
It seems like project url should be in the nuspec and it's being removed?

Also why is the framework reference to mscorlib removed?

File: src/libraries/Directory.Build.targets [hunk 2]
```diff
@@ -272,4 +273,20 @@
       </ItemGroup>
     </When>
   </Choose>
+
+  <Target Name="AddLibIntelDocs">
```
Comment (safern) [inline src/libraries/Directory.Build.targets:277]: Cool this will make it very simple to hook the hybrid mode for docs in between intellisense and triple slash.

Also, could you rename to LibIntellisenseDocs? That's more clear.

File: src/libraries/Directory.Build.targets [hunk 3]
```diff
@@ -272,4 +273,20 @@
...
+   <ItemGroup>
+      <TfmSpecificPackageFile Include="$(XmlDocFileRoot)/1033/$(TargetName).xml" Condition="Exists('$(XmlDocFileRoot)/1033/$(TargetName).xml')">
+        <PackagePath>lib/$(TargetFramework)</PackagePath>
+      </TfmSpecificPackageFile>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="RemoveXmlFilesFromBuildOutput" BeforeTargets="GenerateNuspec">
```
Comment (safern) [inline src/libraries/Directory.Build.targets:285]: Is this because we want to use the ones from the intellisense package?

Comment (Anipik) [reply src/libraries/Directory.Build.targets:285]: yes thats correct.

**Comment by Anipik** [comment]
> It seems like project url should be in the nuspec and it's being removed?

It was being set to an empty, fixed and checked nuspec it contains the same value now

Also why is the framework reference to mscorlib removed?

> sdk adds this by default so packages dont need to have this anymore.

File: src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/Microsoft.Extensions.Configuration.EnvironmentVariables.csproj [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <IsPackable>true</IsPackable>
+    <IncludeSymbols>true</IncludeSymbols>
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/Microsoft.Extensions.Configuration.EnvironmentVariables.csproj:6]: Do you plan to set this property for every csproj? Why can't it be set in a central place?

File: Directory.Build.props [hunk 5]
```diff
@@ -232,7 +232,9 @@
     <Owners>microsoft,dotnetframework</Owners>
     <RuntimeIdGraphDefinitionFile>$([MSBuild]::NormalizePath('$(LibrariesProjectRoot)', 'Microsoft.NETCore.Platforms', 'src', 'runtime.json'))</RuntimeIdGraphDefinitionFile>
     <LicenseFile>$(MSBuildThisFileDirectory)LICENSE.TXT</LicenseFile>
+    <PackageIconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</PackageIconUrl>
```
[the remark this answers is not in the document]
Comment (ViktorHofer) [reply Directory.Build.props:235]: That property is already deprecated.

File: src/libraries/libraries-packages.proj (hunk 1 above)
Comment (Anipik) [reply src/libraries/libraries-packages.proj:4]: Right, i might have missed something.

**Comment by Anipik** [comment]
cc headsup @dseefeld  this change might effect on how u build the packages. let me know if it breaks anything and i can help

File: Directory.Build.props [hunk 6]
```diff
@@ -230,9 +230,11 @@
     <RepositoryUrl>https://github.com/dotnet/$(GitHubRepositoryName)</RepositoryUrl>
     <PackageProjectUrl>https://dot.net</PackageProjectUrl>
     <Owners>microsoft,dotnetframework</Owners>
+    <IncludeSymbols>true</IncludeSymbols>
```
Comment (ViktorHofer) [inline Directory.Build.props:233]: Does this global setting effect other packages? We do have a few others that are already NuGet Pack task based. What about the existing pkgproj ones?

File: eng/packaging.props [hunk 7]
```diff
@@ -27,6 +27,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
+    <NuSpecOutputPath Condition="'$(NuSpecOutputPath)' == ''">$([MSBuild]::NormalizeDirectory('$(ArtifactsPackagesDir)', 'specs'))</NuSpecOutputPath>
```
Comment (ViktorHofer) [inline eng/packaging.props:30]: Why do we set this for libraries packages only? Should this be moved into the root Directory.Build files instead?

Do we need the condition?

File: src/libraries/Directory.Build.props [hunk 8]
```diff
@@ -29,7 +29,7 @@
     <BuildTargetFramework Condition="'$(BuildTargetFramework)' == '' and '$(TargetFramework)' != ''">$([System.Text.RegularExpressions.Regex]::Replace('$(TargetFramework)', '(-[^;]+)', ''))</BuildTargetFramework>
     <!-- Build all .NET Framework configurations when net48 is passed in. This is for convenience. -->
     <AdditionalBuildTargetFrameworks Condition="'$(BuildTargetFramework)' == 'net48'">net45;net451;net452;net46;net461;net462;net47;net471;net472</AdditionalBuildTargetFrameworks>
-    <AdditionalBuildTargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true' and '$(BuildAllProjects)' == 'true'">$(AdditionalBuildTargetFrameworks);netstandard2.0</AdditionalBuildTargetFrameworks>
+    <AdditionalBuildTargetFrameworks Condition="'$(DotNetBuildFromSource)' == 'true' and '$(BuildAllProjects)' == 'true'">$(AdditionalBuildTargetFrameworks);net461;netstandard2.0</AdditionalBuildTargetFrameworks>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.props:32]: IIRC (@ericstj likely has more context) we only support building .NETCoreApp current and .NETStandard builds when building from source. Why do we require .NET Framework assets to be built as well with the switch from pkgproj to csproj?

File: src/libraries/Directory.Build.targets [hunk 9]
```diff
@@ -272,4 +273,20 @@
...
+    <PropertyGroup>
+      <AllowedOutputExtensionsInPackageBuildOutputFolder>.dll;.exe;.winmd;.json;.pri;</AllowedOutputExtensionsInPackageBuildOutputFolder>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets:287]: You want to add a comment with a TODO to remove this after https://github.com/NuGet/NuGet.Client/pull/3980 is in.

File: src/libraries/Directory.Build.targets [hunk 10]
```diff
@@ -12,6 +12,7 @@
   <Import Project="..\..\Directory.Build.targets" />
 
   <PropertyGroup>
+    <TargetsForTfmSpecificContentInPackage Condition="'$(IsPackable)' == 'true'">LibIntellisenseDocs;$(TargetsForTfmSpecificContentInPackage)</TargetsForTfmSpecificContentInPackage>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets:15]: nit: I would rather append than prepend to the list but you decide :)

```suggestion
    <TargetsForTfmSpecificContentInPackage Condition="'$(IsPackable)' == 'true'">$(TargetsForTfmSpecificContentInPackage);LibIntellisenseDocs</TargetsForTfmSpecificContentInPackage>
```

**Comment by ViktorHofer** [comment]
Please move the PackageDescription into the csproj from the Directory.Build.props as well: https://github.com/dotnet/runtime/blob/1c9e2009d9942a6df31daad8c912c68446856fa2/src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/Directory.Build.props#L4.

File: Directory.Build.props (hunk 6 above)
Comment (Anipik) [reply Directory.Build.props:233]: > What about the existing pkgproj ones, especially coreclr, mono and installer ones.

They don't use this property. They have their own set of properties.


> Does this global setting affect other packages? We do have a few others that are already NuGet Pack task based.

i restored them to the state they were before this change, for some of them we might want to revisit on why they are not publishing symbols.

File: eng/packaging.props (hunk 7 above)
Comment (Anipik) [reply eng/packaging.props:30]: Currently they donot produce that, we can fix in that in a separate change.

File: src/libraries/Directory.Build.props (hunk 8 above)
Comment (Anipik) [reply src/libraries/Directory.Build.props:32]: We build the .NETFramework assets indirectly throught package projs. Sourcebuild produces packages in the vertical builds so we will need this config.

Comment (ViktorHofer) [reply src/libraries/Directory.Build.props:32]: I don't understand what you mean by that. Please elaborate

File: eng/packaging.props (hunk 7 above)
Comment (ViktorHofer) [reply eng/packaging.props:30]: If for other packages we don't need the nuspec in a specific place, I'm fine with that. Out of curiosity, what are using the nuspec for? Package validation?

Comment (Anipik) [reply eng/packaging.props:30]: i dont think we need them in a specific place place, they are not used in anything

File: src/libraries/Directory.Build.props (hunk 8 above)
Comment (Anipik) [reply src/libraries/Directory.Build.props:32]: Source build doesn't have an all config leg. It builds packages in a vertical build. During the build part, it builds the netcoreapp and netstandard assets. During pack task, it builds pkgprojs which have project references to the csprojs and that results in building all the config (which includes the .NETFramework config)

Comment (ViktorHofer) [reply src/libraries/Directory.Build.props:32]: > It builds packages in a vertical build.

If source build is running as part of a vertical build, I wouldn't expect any .NETFrameworks in projects to be built.

> During pack task, it builds pkgprojs which have project references to the csprojs and that results in building all the config

We filter project references to only build for the applicable `BuildTargetFramework`. Why doesn't it filter?

File: eng/packaging.props (hunk 7 above)
Comment (ViktorHofer) [reply eng/packaging.props:30]: Then why is this change necessary? Can't we just have them in their default location which I believe is in their intermediate output directories?

Comment (Anipik) [reply eng/packaging.props:30]: > Then why is this change necessary? Can't we just have them in their default location which I believe is in their intermediate output directories?

just to keep the things similar to the other nuspecs, we can move them all together to their default locations

File: src/libraries/Directory.Build.props (hunk 8 above)
Comment (Anipik) [reply src/libraries/Directory.Build.props:32]: Because its not being set, its value is ""

Comment (ViktorHofer) [reply src/libraries/Directory.Build.props:32]: Why is that so?

Comment (Anipik) [reply src/libraries/Directory.Build.props:32]: this is the only place we set it and this wont get executed https://github.com/dotnet/runtime/blob/main/src/libraries/Directory.Build.props#L29

Comment (ViktorHofer) [reply src/libraries/Directory.Build.props:32]: Let me rephrase this discussion. I don't yet understand how changing the way nuget packages are product could result in an addition of compile time assets. How was it possible to not include .NET Framework assets in pkgprojs and why isn't that anymore possible with the NuGet Pack task on i.e. csprojs?

File: eng/packaging.props (hunk 7 above)
Comment (ViktorHofer) [reply eng/packaging.props:30]: Makes sense, let's post pone this.

File: Directory.Build.props (hunk 6 above)
Comment (ViktorHofer) [reply Directory.Build.props:233]: Thanks for double checking. As I'm still a bit worried about this, cc'ing @ericstj in case he has more opinions on this change.

File: src/libraries/Directory.Build.props (hunk 8 above)
Comment (Anipik) [reply src/libraries/Directory.Build.props:32]: > s. How was it possible to not include .NET Framework assets in pkgprojs and

pkgproj pack doesnt report missing files as and error and we didnt check the sourcebuild packages to see if they contain all the assets.

>  how changing the way nuget packages are product could result in an addition of compile time assets.

Nuget pack does report this as error if there are missing files 


> why isn't that anymore possible with the NuGet Pack task on i.e. csprojs

We might be able to suppress that, but i talked with @dseefeld and they are already building the netfx config in the sourcebuild repo with this  https://github.com/dotnet/source-build/blob/release/5.0/patches/runtime/0033-Build-net4-in-source-build.patch, so i think we should follow the same thing

File: src/libraries/Directory.Build.targets [hunk 11]
```diff
@@ -229,6 +230,11 @@
     <ExcludeFromPackage Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', '$(NetCoreAppCurrent)')) and '$(ExcludeCurrentNetCoreAppFromPackage)' == 'true'">true</ExcludeFromPackage>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(DotnetBuildFromSource)' == 'true' and '$(TargetFramework)' != ''">
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets:233]:
```suggestion
  <PropertyGroup Condition="'$(DotnetBuildFromSource)' == 'true' and '$(IsPackable)' == 'true'">
```

File: src/libraries/Directory.Build.targets [hunk 12]
```diff
@@ -229,6 +230,11 @@
...
+    <IncludeBuildOutput Condition="!$(TargetFramework.StartsWith('$(NetCoreAppCurrent)')) and '$(TargetFramework)' != 'netstandard2.0'">false</IncludeBuildOutput>
+    <SuppressDependenciesWhenPacking Condition="!$(TargetFramework.StartsWith('$(NetCoreAppCurrent)')) and '$(TargetFramework)' != 'netstandard2.0'">true</SuppressDependenciesWhenPacking>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets:235]:
```suggestion
    <IncludeBuildOutput>false</IncludeBuildOutput>
    <SuppressDependenciesWhenPacking>true</SuppressDependenciesWhenPacking>
```

File: src/libraries/Directory.Build.targets [hunk 13]
```diff
@@ -272,4 +278,21 @@
       </ItemGroup>
     </When>
   </Choose>
+
+  <Target Name="LibIntellisenseDocs" Condition="'$(IncludeBuildOutput)' == 'true'">
+   <ItemGroup>
+      <TfmSpecificPackageFile Include="$(XmlDocFileRoot)/1033/$(TargetName).xml" Condition="Exists('$(XmlDocFileRoot)/1033/$(TargetName).xml')">
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets:284]: nit: we standardized on `\` slashes.

```suggestion
      <TfmSpecificPackageFile Include="$(XmlDocFileRoot)1033\$(TargetName).xml" Condition="Exists('$(XmlDocFileRoot)1033\$(TargetName).xml')">
```

File: src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/Microsoft.Extensions.Configuration.EnvironmentVariables.csproj [hunk 14]
```diff
@@ -2,7 +2,10 @@
 
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
+    <IsPackable>true</IsPackable>
     <EnableDefaultItems>true</EnableDefaultItems>
+    <NoWarn>$(NoWarn);NU5128</NoWarn>
```
Comment (ViktorHofer) [inline src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/Microsoft.Extensions.Configuration.EnvironmentVariables.csproj:7]: Why is this necessary? I would not expect to have a file present for a tfm without a dependency node.

Comment (Anipik) [reply src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/Microsoft.Extensions.Configuration.EnvironmentVariables.csproj:7]: we can remove this now, earlier it was requires when i was not removing the dependency node

Comment (Anipik) [reply src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/Microsoft.Extensions.Configuration.EnvironmentVariables.csproj:7]: we can remove this now, earlier it was requires when i was not removing the dependency node

Comment (ViktorHofer) [reply src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/Microsoft.Extensions.Configuration.EnvironmentVariables.csproj]: Let's please agree on that we never accept this NoWarn to be set as it indicates a wrong use of the Pack task.

File: src/libraries/Directory.Build.targets [hunk 15]
```diff
@@ -272,4 +278,21 @@
       </ItemGroup>
     </When>
   </Choose>
+
+  <Target Name="LibIntellisenseDocs" Condition="'$(IncludeBuildOutput)' == 'true'">
+   <ItemGroup>
+      <TfmSpecificPackageFile Include="$(XmlDocFileRoot)1033\$(TargetName).xml" Condition="Exists('$(XmlDocFileRoot)1033\$(TargetName).xml')">
+        <PackagePath>lib/$(TargetFramework)</PackagePath>
+      </TfmSpecificPackageFile>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets:286]: nit:

```suggestion
      <TfmSpecificPackageFile Include="$(XmlDocFileRoot)1033\$(TargetName).xml"
                              Condition="Exists('$(XmlDocFileRoot)1033\$(TargetName).xml')"
                              PackagePath="lib/$(TargetFramework)" />
```

Comment (ViktorHofer) [reply src/libraries/Directory.Build.targets:286]: hard to apply the suggestion with the right whitespaces. Hope this is correct :D

**Review by ViktorHofer — APPROVED** [review]
I'm now very happy about this PR 👍 Thanks

**Comment by Anipik** [comment]
awesome

**Comment by Anipik** [comment]
failed test https://github.com/dotnet/runtime/issues/52031
